### PR TITLE
[Feature] support iceberg manifest cache

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What I'm doing:
1. enable iceberg meta cache by default
2. support iceberg manifest cache on job planning

TODO: support distributed plan in the StarRocksIcebergTableScan

Fixes #issue
https://github.com/StarRocks/starrocks/issues/43460

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/iceberg/DeleteFileIndex.java [hunk 1]
```diff
@@ -325,6 +328,8 @@ static class Builder {
         private boolean caseSensitive = true;
         private ExecutorService executorService = null;
         private ScanMetrics scanMetrics = ScanMetrics.noop();
+        private Cache<String, Set<DeleteFile>> deleteFileCache;
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/org/apache/iceberg/DeleteFileIndex.java:331]: are we going to rewrite iceberg implementation?

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java [hunk 2]
```diff
@@ -51,21 +53,29 @@ public class CachingIcebergCatalog implements IcebergCatalog {
     private final ExecutorService backgroundExecutor;
 
     private final IcebergCatalogProperties icebergProperties;
+    private final Cache<String, Set<DataFile>> dataFileCache;
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java:56]: key is manifest file path?

**Review by dirtysalt — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java:56]: yes

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java [hunk 3]
```diff
@@ -51,21 +53,29 @@ public class CachingIcebergCatalog implements IcebergCatalog {
...
+    private final Cache<String, Set<DeleteFile>> deleteFileCache;
 
     public CachingIcebergCatalog(String catalogName, IcebergCatalog delegate, IcebergCatalogProperties icebergProperties,
                                  ExecutorService executorService) {
         this.catalogName = catalogName;
         this.delegate = delegate;
         this.icebergProperties = icebergProperties;
+        boolean enableCache = icebergProperties.isEnableIcebergMetadataCache();
         this.databases = newCacheBuilder(icebergProperties.getIcebergMetaCacheTtlSec(),
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java:65]: db cache is useful?

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java:65]: Consistent with before

**Review by Dshadowzh — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java [hunk 4]
```diff
@@ -33,18 +65,271 @@ public static TableScanContext newTableScanContext(Table table) {
...
     }
 
     @Override
     protected TableScan newRefinedScan(Table newTable, Schema newSchema, TableScanContext newContext) {
-        return null;
+        return new StarRocksIcebergTableScan(newTable, newSchema, newContext, scanContext);
+    }
+
+    public StarRocksIcebergTableScan dataFileCache(Cache<String, Set<DataFile>> dataFileCache) {
+        scanContext.setDataFileCache(dataFileCache);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java:90]: why need to set to scanContext

**Review by Youngwb — APPROVED** [review]

**Pull request merged**